sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -348,20 +352,25 @@ case class FileSourceScanExec(
         "DataFilters" -> seqToString(dataFilters),
         "Location" -> locationDesc)
 
-    val withSelectedBucketsCount = relation.bucketSpec.map { spec =>
-      val numSelectedBuckets = optionalBucketSet.map { b =>
-        b.cardinality()
+    // TODO(SPARK-32986): Add bucketed scan info in explain output of FileSourceScanExec
+    if (bucketedScan) {
+      val withSelectedBucketsCount = relation.bucketSpec.map { spec =>
```

Comment:
nit: instead of writing
```
val withSelectedBucketsCount =  some_code
withSelectedBucketsCount
```
we can just write `some_code`

Reply:
@cloud-fan - updated.